src/main/java/io/zonky/test/db/postgres/junit5/PreparedDbExtension.java

```diff
@@ -17,18 +17,17 @@
 import io.zonky.test.db.postgres.embedded.DatabasePreparer;
 import io.zonky.test.db.postgres.embedded.EmbeddedPostgres;
 import io.zonky.test.db.postgres.embedded.PreparedDbProvider;
-import org.junit.jupiter.api.extension.AfterAllCallback;
-import org.junit.jupiter.api.extension.BeforeAllCallback;
-import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.*;
 
 import javax.sql.DataSource;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.function.Consumer;
 
-public class PreparedDbExtension implements BeforeAllCallback, AfterAllCallback {
+public class PreparedDbExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback, AfterEachCallback {
 
     private final DatabasePreparer preparer;
+    private boolean perClass = false;
```

Comment:
How does this work? How can I change the parameter to refresh the database before each test?

Maybe this behavior could be user configurable. And also an enum type would be better than the boolean.

```java
@RegisterExtension
public PreparedDbExtension db = EmbeddedPostgresExtension
        .preparedDatabase(FlywayPreparer.forClasspathLocation("db/testing"))
        .refresh(RefreshMode.BEFORE_EACH_TEST);
```

What do you think?

Reply:
Well the idea was setting static makes it per class while non-static is per-test.

Reply:
Ok, now it makes sense. So could you please add some tests that will verify this behavior? Because the extension is currently registered in all tests as non-static, and all test classes have only a single test method. So the database refresh feature is not tested at all at this time.

Reply:
Test that fails without this change added.